pkg/prowgen/prowgen.go

```diff
@@ -207,61 +208,113 @@ func runOpenShiftReleaseGenerator(ctx context.Context, openShiftRelease Reposito
 	return nil
 }
 
-func injectSlackReporterConfig(inConfig *Config, openShiftRelease Repository) error {
-	// Inject Slack reporter for each repository and branch
+func runJobConfigInjectors(inConfig *Config, openShiftRelease Repository) error {
+	injectors := JobConfigInjectors{
+		alwaysRunInjector(),
+		slackInjector(),
+	}
+	return injectors.Inject(inConfig, openShiftRelease)
+}
+
+func slackInjector() JobConfigInjector {
+	return JobConfigInjector{
+		Type: Periodic,
+		Update: func(r *Repository, jobConfig *prowconfig.JobConfig) error {
+			for i := range jobConfig.Periodics {
+				jobConfig.Periodics[i].ReporterConfig = &prowapi.ReporterConfig{
+					Slack: &prowapi.SlackReporterConfig{
+						Channel: r.SlackChannel,
+						JobStatesToReport: []prowapi.ProwJobState{
+							prowapi.SuccessState,
+							prowapi.FailureState,
+							prowapi.ErrorState,
+						},
+						ReportTemplate: `{{if eq .Status.State "success"}} :rainbow: Job *{{.Spec.Job}}* ended with *{{.Status.State}}*. <{{.Status.URL}}|View logs> :rainbow: {{else}} :volcano: Job *{{.Spec.Job}}* ended with *{{.Status.State}}*. <{{.Status.URL}}|View logs> :volcano: {{end}}`,
+					},
+				}
+			}
+			return nil
+		},
+	}
+}
 
-	log.Println("Injecting Slack reporter configs")
+func alwaysRunInjector() JobConfigInjector {
+	return JobConfigInjector{
+		Type: PreSubmit,
+		Update: func(r *Repository, jobConfig *prowconfig.JobConfig) error {
+			tests, err := discoverE2ETests(*r)
+			if err != nil {
+				return fmt.Errorf("failed to discover tests: %w", err)
+			}
+
+			for k := range jobConfig.PresubmitsStatic {
+				for i := range jobConfig.PresubmitsStatic[k] {
+					if err != nil {
+						return err
+					}
+
+					variant := jobConfig.PresubmitsStatic[k][i].Labels["ci-operator.openshift.io/variant"]
+					ocpVersion := strings.SplitN(variant, "-", 2)[0]
+
+					for _, t := range tests {
+						name := ToName(*r, &t, ocpVersion)
+						if t.OnDemand && name == jobConfig.PresubmitsStatic[k][i].Name {
```

Comment:
```suggestion
						if t.OnDemand && strings.Contains(jobConfig.PresubmitsStatic[k][i].Name, name) {
```

Reply:
The name is `name: pull-ci-openshift-knative-eventing-hyperfoil-benchmark-main-412-test-kafka-8293723-aws-ocp-412` so `==` doesn't work

Reply:
^^ @aliok